core/state_transition.go

```diff
@@ -532,7 +532,7 @@ func (st *stateTransition) execute() (*ExecutionResult, error) {
 
 	effectiveTip := msg.GasPrice
 	if rules.IsLondon {
-		effectiveTip = new(big.Int).Sub(msg.GasFeeCap, st.evm.Context.BaseFee)
+		effectiveTip = new(big.Int).Sub(msg.GasPrice, st.evm.Context.BaseFee)
 		if effectiveTip.Cmp(msg.GasTipCap) > 0 {
 			effectiveTip = msg.GasTipCap
```

Comment:
Can this clause even happen? `GasPrice` is the effective price paid. In case of EIP-1559, it is a computed value based on `GasFeeCap` and `GasTipCap`.

Reply:
I'm not sure what exactly you mean by "Can this clause even happen? ", this clause will happen when `rules.IsLondon` is `true`, guess you mean something else?

Reply:
Yes, the clause below is unnecessary, I was planning to remove it after this PR is merged. Do you want to remove it in this PR?